### PR TITLE
fix(wallet): added ChainID to balance cache, as now it is shared between

### PR DESCRIPTION
**Pull request by IvanBelyakoff**
services and contains balances for all addresses and chains.
Made rpc chain client return ChainID property on `NetworkID()` method

Updates #3835

### CODE REVIEW

File: services/wallet/balance/balance_cache.go [hunk 1]
```diff
@@ -35,145 +36,176 @@ type Cacher interface {
...
-				maxBlock = key
+		for chainID, cache := range chainCache {
+			if len(cache) == 0 {
+				continue
 			}
-			if key < minBlock {
-				minBlock = key
+
+			var maxBlock uint64 = 0
+			var minBlock uint64 = 18446744073709551615
```
Comment (stefandunca) [inline services/wallet/balance/balance_cache.go]: I wonder where is this constant coming from. Can it be sourced for example, from MaxUint64 from a reliable source?

File: services/wallet/balance/balance_cache.go [hunk 2]
```diff
@@ -35,145 +36,176 @@ type Cacher interface {
...
-		var minBlock uint64 = 18446744073709551615
-		for key := range cache {
-			if key > maxBlock {
-				maxBlock = key
-			}
-			if key < minBlock {
-				minBlock = key
+		for chainID, cache := range chainCache {
+			var maxBlock uint64 = 0
+			var minBlock uint64 = 18446744073709551615
```
Comment (stefandunca) [inline services/wallet/balance/balance_cache.go]: Share this magic number, maybe?

**Review by stefandunca — APPROVED** [review]
I left some questions inline in case you can clarify them.

**Review by saledjenic — APPROVED** [review]

File: services/wallet/balance/balance_cache.go (hunk 1 above)
Comment (IvanBelyakoff) [reply services/wallet/balance/balance_cache.go]: @stefandunca Good point! Fixed.

File: services/wallet/balance/balance_cache.go (hunk 2 above)
Comment (IvanBelyakoff) [reply services/wallet/balance/balance_cache.go]: Fixed.

**Pull request merged**